controllers/vcluster_controller.go

```diff
@@ -280,51 +334,25 @@ func (r *VClusterReconciler) redeployIfNeeded(_ context.Context, vCluster *v1alp
 		chartName = constants.DefaultVClusterChartName
 	}
 
-	if vCluster.Spec.VirtualClusterVersion == nil || *vCluster.Spec.VirtualClusterVersion == "" {
-		return fmt.Errorf("empty value of the .spec.virtualClusterVersion field")
+	if vCluster.Spec.HelmRelease == nil || vCluster.Spec.HelmRelease.Chart.Version == "" {
+		return fmt.Errorf("empty value of the .spec.HelmRelease.Version field")
 	}
 	// chart version
-	chartVersion := *vCluster.Spec.VirtualClusterVersion
+	chartVersion := vCluster.Spec.HelmRelease.Chart.Version
 
 	if len(chartVersion) > 0 && chartVersion[0] == 'v' {
 		chartVersion = chartVersion[1:]
 	}
 
-	if vCluster.Spec.KubernetesVersion == nil || *vCluster.Spec.KubernetesVersion == "" {
-		return fmt.Errorf("empty value of the .spec.kubernetesVersion field")
-	}
-	kubernetesVersionStr := strings.Split(*vCluster.Spec.KubernetesVersion, ".")
-	if len(kubernetesVersionStr) < 2 || len(kubernetesVersionStr) > 3 {
-		return fmt.Errorf("invalid value of the .spec.kubernetesVersion field: %s", *vCluster.Spec.KubernetesVersion)
-	}
-	if len(kubernetesVersionStr) == 3 {
-		r.Log.Info("kubernetes patch version defined in .spec.kubernetesVersion field will be ignored, latest supported patch version will be used",
-			"namespace", vCluster.Namespace,
-			"clusterName", vCluster.Name,
-		)
-	}
-	kubernetesVersion := vclusterhelm.Version{
-		Major: kubernetesVersionStr[0],
-		Minor: kubernetesVersionStr[1],
+	if vCluster.Spec.HelmRelease == nil || vCluster.Spec.HelmRelease.Values == "" {
```

Comment:
No need to require values, people should be able to deploy it without values

Reply:
done